### ISSUE TITLE
Effect of worker thread on the loss during training

### ISSUE THREAD

**Issue by grv1207** [facebookresearch/fastText#665]
I noticed one strange behavior that increases the thread count from 1 to any count, increases the loss during training, below is the detail description: I was expecting the loss in some range but there is a there is a noticeable difference between thread at 1 and other threads. However, the loss at higher thread count is in the same range. Can someone explain this behavior..?

loss | thread
-- | --
1.443683 | 1
2.306886 | 3
2.328125 | 5
2.29902 | 8
2.28799 | 10
2.293041 | 15

**Comment by Celebio** [comment]
Hi. The displayed loss is not the loss you might get if you do a forward pass on your training set. Instead, it is an average over all samples during the **whole** training. At the beginning of the training the loss at each thread will be high, that's why you observe an increasing averaged loss as a function of number of threads.

**Comment by Celebio** [comment]
Hi @grv1207,
I hope this answers your question, I will now close the issue. If you have any doubts or further questions you can reopen the issue or ask your questions to our community on the [facebook group](https://www.facebook.com/groups/1174547215919768/).

Regards,
Onur
